### PR TITLE
Mongokitten4/Mongo-Drive update

### PR DESCRIPTION
**Pull request by cam25**
Updated Mongo-Drive to be compatible with latest version of Fluent/Vapor

### CODE REVIEW

File: Sources/Query+MongoKitten.swift [hunk 1]
```diff
@@ -3,16 +3,15 @@ import Fluent
...
 
-        let aqts = try filters.map { try $0.makeAQT() }
-
+        let aqts = try filters.map { try $0.wrapped?.makeAQT() }        
         if aqts.isEmpty {
             return .nothing
         }
 
-        return .and(aqts)
+        return .and(aqts as! [AQT])
```
Comment (vzsg) [inline Sources/Query+MongoKitten.swift]: Don't use a force cast here!
Instead, try rewriting the above line to:

```swift
let aqts = try filters.flatMap { try $0.wrapped?.makeAQT() } // aqts is now an [AQT]
```

File: Sources/MongoDriver.swift [hunk 2]
```diff
@@ -2,7 +2,18 @@ import Foundation
 import Fluent
 import MongoKitten
 
-public class MongoDriver: Fluent.Driver {
+public class MongoDriver: Connection, Fluent.Driver{
+    public /// The naming convetion to use for foreign
```
Comment (vzsg) [inline Sources/MongoDriver.swift]: What's up with this `public` keyword here?

File: Sources/Filter+MongoKitten.swift [hunk 3]
```diff
@@ -52,16 +59,16 @@ extension Fluent.Filter {
                 query = .and(ands)
             }
         case .group(let relation, let filters):
-            let aqts = try filters.map { try $0.makeAQT() }
+            let aqts = try filters.map { try $0.wrapped?.makeAQT()}
```
Comment (vzsg) [inline Sources/Filter+MongoKitten.swift:62]: Same as the other, use flatMap instead of force casting.

File: Sources/Filter+MongoKitten.swift [hunk 4]
```diff
@@ -26,11 +31,13 @@ extension Fluent.Filter {
             case .notEquals:
                 query = .valNotEquals(key: key, val: val.bson)
             case .contains:
-                query = .contains(key: key, val: val.bson.string)
+                query = .containsElement(key: key, match: val.bson as! AQT)
```
Comment (vzsg) [inline Sources/Filter+MongoKitten.swift]: Are you sure that this is the right way?

**Review by vzsg — CHANGES_REQUESTED** [review]
I added a few observations of mine, but this is by no means conclusive.

File: Sources/MongoDriver.swift (hunk 2 above)
Comment (cam25) [reply Sources/MongoDriver.swift]: Should be removed. Not serving any purpose here.

File: Sources/Query+MongoKitten.swift (hunk 1 above)
Comment (cam25) [reply Sources/Query+MongoKitten.swift]: Will update, thanks!

File: Sources/Filter+MongoKitten.swift (hunk 3 above)
Comment (cam25) [reply Sources/Filter+MongoKitten.swift:62]: Done

**Comment by Joannis** [comment]
I think we should remove the `.swift-version` file by now

**Comment by Joannis** [comment]
This driver will be Swift 3.1 and Vapor 2 only. So this PR will at least need to be merged after Vapor 2 gets out of Beta and MongoKitten 4 has adapted to that.

**Comment by vzsg** [comment]
There's a lot of work before this can be released in the first place. But if it starts working, I don't see an issue – all Vapor repos have their beta versions in the master branch.

**Pull request closed without merging**